fineract-provider/src/main/java/org/apache/fineract/portfolio/note/domain/Note.java

```diff
@@ -119,7 +121,7 @@ public Note(final Client client, final String note) {
         this.noteTypeId = NoteType.CLIENT.getValue();
     }
 
-    private Note(final Group group, final String note) {
+    public Note(final Group group, final String note) {
```

Comment:
Use @RequiredArgsConstructor. Let's be consistent.

Reply:
There's another PR, #4020, that addresses the `Note` instantiation issues by implementing the **Factory** pattern. 
You're welcome to review it.